solidity/test/InterchainAccountRouter.t.sol

```diff
@@ -867,4 +867,65 @@ contract InterchainAccountRouterTest is InterchainAccountRouterTestBase {
             customHook
         );
     }
+
+    function testFuzz_callRemoteCommitReveal(bytes32 commitment) public {
+        // act
+        originIcaRouter.callRemoteCommitReveal(
+            destination,
+            routerOverride,
+            ismOverride,
+            bytes(""),
+            new TestPostDispatchHook(),
+            bytes32(0),
+            commitment
+        );
+
+        // Process message
+        environment.processNextPendingMessage();
+
+        // assert
+        // Destination ICA router should have the commitment
+        assertEq(
+            address(destinationIcaRouter.verifiedCommitments(commitment)),
+            address(ica)
+        );
+    }
+
+    function testFuzz_revealAndExecute(
+        bytes32 data,
+        uint256 value,
+        bytes32 salt
+    ) public {
+        // Arrange
+        CallLib.Call[] memory calls = getCalls(data, value);
+        bytes32 commitment = keccak256(abi.encode(salt, calls));
+        deal(address(ica), value); // Ensure ICA has enough balance to execute calls
+
+        // Act
+        originIcaRouter.callRemoteCommitReveal(
+            destination,
+            routerOverride,
+            ismOverride,
+            bytes(""),
+            new TestPostDispatchHook(),
+            bytes32(0),
+            commitment
+        );
+        // Process commit message
+        environment.processNextPendingMessage();
+
+        // Destination ICA router should have the commitment after commit message
+        assertEq(
+            address(destinationIcaRouter.verifiedCommitments(commitment)),
+            address(ica)
+        );
+
+        // Process reveal message
+        environment.processNextPendingMessage();
+        destinationIcaRouter.revealAndExecute(calls, salt);
```

Comment:
the `processNextPendingMessage` should do the execution and then the `revealAndExecute` should revert

Reply:
Updated